uaclient/cli.py

```diff
@@ -1325,6 +1326,18 @@ def main(sys_argv=None):
     logging.debug(
         util.redact_sensitive_logs("Executed with sys.argv: %r" % sys_argv)
     )
+    ua_environment = [
+        "{}={}".format(k, v)
+        for k, v in sorted(os.environ.items())
+        if k.lower() in CONFIG_FIELD_ENVVAR_ALLOWLIST
+        or k.startswith("UA_FEATURES")
+        or k == "UA_CONFIG_FILE"
+    ]
+    logging.debug(
+        util.redact_sensitive_logs(
+            "Executed with UA environment variables: %r" % ua_environment
+        )
+    )
```

Comment:
Let's keep logging to minimum and only log if we have something significant to say. This is a problem with `cloud-init` and a waaay too busy log, Reduced logging will keep our log files more human-friendly.

```suggestion
    if ua_environment:
        logging.debug(
            util.redact_sensitive_logs(
                "Executed with UA environment variables: %r" % ua_environment
            )
        )
```

Reply:
It amazes me when empty lists evaluate to false, I guess I don't miss JS

(just a little)